### PR TITLE
SupportBee #3797

### PR DESCRIPTION
**Pull request by lcaresia**

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/supportbee/sources/common/common.mjs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    while (true) {
+      const resources = await this.getResources({
+        params: {
+          per_page: 100,
+          page,
+        },
+      });
+
+      resources.reverse().forEach(this.emitEvent);
+      this._setLastResourceId(resources[0].id);
```
Comment (jcortes) [inline components/supportbee/sources/common/common.mjs:55]: What happens if `resources` is empty `[]`? I think here you should check that right? something like
```suggestion
      if (resources.length) {
        this._setLastResourceId(resources[0].id);
      }
```

File: components/supportbee/sources/common/common.mjs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+      if (
+        resources.length < 100 ||
+        resources.filter((payment) => payment.id === lastResourceId)
```
Comment (jcortes) [inline components/supportbee/sources/common/common.mjs:59]:
```suggestion
        resources.find((payment) => payment.id === lastResourceId)
```

**Review by jcortes — CHANGES_REQUESTED** [review]
Hi @lcaresia I've just added a few request changes and observations. Thanks!

File: components/supportbee/sources/common/common.mjs (hunk 1 above)
Comment (lcaresia) [reply components/supportbee/sources/common/common.mjs:55]: Yes, you are right, sorry for this!

**Review by jcortes — COMMENTED** [review]
Hi @lcaresia LGTM! Ready for QA!

**Comment by DilanAthukorala** [comment]
@lcaresia For "New Comment" source observing the below error when creating source. 
image

**Comment by DilanAthukorala** [comment]
Verification completed and ready for release.

**Review by michelle0927 — APPROVED** [review]

**Pull request merged**